zulipterminal/model.py

```diff
@@ -1205,6 +1205,36 @@ def _handle_update_message_event(self, event: Event) -> None:
                     else:
                         self.index["topics"][stream_id] = []
 
+            if event["propagate_mode"] in ["change_later", "change_all"]:
+                old_subject = event["orig_subject"]
+
+                # If the user is composing a message to this topic, update the topic in
+                # the title_write_box header on a 'change_later' or 'change_all' update
+                # in the topic.
+                if hasattr(self.controller, "view"):
+                    write_box = self.controller.view.write_box
+                    if (
+                        write_box.stream_id == stream_id
+                        and write_box.title_write_box.edit_text == old_subject
+                    ):
+                        write_box.update_topic_compose_header(new_subject)
+
```

Comment:
I've rewritten this code in a way that facilitates a smooth extension when adding support for moving topics between streams, which is why it is a little different from the previous iteration.